File: classic puzzle-medium/minimal-number-of-swaps.rb
#https://www.codingame.com/training/medium/minimal-number-of-swaps
#MINIMAL NUMBER OF SWAPS
n=gets
m=gets.split.map(&:to_i)
s=0
a=m.sort.reverse.each_with_index{|x,i|
  if x!=m[i]
    s+=1
  end
}
puts s/2

File: classic puzzle-medium/minimal-number-of-swaps.py
#https://www.codingame.com/training/medium/minimal-number-of-swaps
#MINIMAL NUMBER OF SWAPS
import sys
import math

n = int(input())
x=[int(i) for i in input().split()]
a=sorted(x.copy(),reverse=True)
cnt=0
for i in range(n):
    if a[i]!=x[i]:
        cnt+=1
print(cnt//2)
